README.md

```diff
@@ -108,24 +119,127 @@ The compiled workflow can be executed via the DNAnexus command line client or we
 
 ![this](doc/bam_chrom_counter.png)
 
-### Strict syntax
+### Validate the workflow
 
 dxCompiler uses [wdlTools](https://github.com/dnanexus/wdlTools), a parser that adheres strictly to the WDL specifications. Most of the problematic automatic type conversions that are allowed by some other WDL runtime engines are not allowed by dxCompiler. Please use the command line tools in wdlTools (e.g. `check` and `lint`) to validate your WDL files before trying to compile them with dxCompiler.
 
 ## CWL
 
-To compile a CWL workflow/tool, you will first need to process it using the following steps:
+### Preprocess CWL workflow
+
+dxCompiler requires the source CWL file be "packed" as a `cwl.json` file, which contains a single compound workflow with all the dependent processes included.
+Before compiling your CWL workflow/tool, you might first need to process it into the packed format using the following steps:
```

Comment:
How about merging these two sentence into one, for example:

dxCompiler requires the source CWL file to be "packed" as a `cwl.json` file, which contains a single compound workflow with all the dependent processes included. Additionally, you may need to upgrade the version of your workflow to 1.2. The following examples demonstrate how to do that:

1. Install ...